subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -754,6 +754,22 @@ static int wifi_get_version(uint32_t mgmt_request, struct net_if *iface,
 
 NET_MGMT_REGISTER_REQUEST_HANDLER(NET_REQUEST_WIFI_VERSION, wifi_get_version);
 
+static int wifi_btm_query(uint32_t mgmt_request, struct net_if *iface,
+			   void *data, size_t len)
+{
+	const struct device *dev = net_if_get_device(iface);
+	const struct wifi_mgmt_ops *const wifi_mgmt_api = get_wifi_api(iface);
+	uint8_t query_reason = *((uint8_t *)data);
+
+	if (wifi_mgmt_api == NULL || wifi_mgmt_api->btm_query == NULL) {
+		return -ENOTSUP;
+	}
+
+	return wifi_mgmt_api->btm_query(dev, query_reason);
```

Comment:
should also check for valid reason (based on enum) before calling driver API? or do you want allow arbitrary reasons and let implementation handle this?

Reply:
If you move the check from shell->mgmt then it covers both shell and API cases. Also, as said above check covers the full range min-max, but we only have three enumerations, so, need to extend check or add a enumeration for UNKNOWN_REASON and encode anyother values as UNKNOWN_REASON.

Reply:
Updated as comment